fe2-android/app/src/main/java/com/github/dedis/popstellar/model/objects/Channel.java

```diff
@@ -0,0 +1,132 @@
+package com.github.dedis.popstellar.model.objects;
+
+import androidx.annotation.NonNull;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * This class represent a channel used to subscribe to a certain part of the messages. More detailed
+ * information are given in the protocol specification of the application.
+ *
+ * <p>It is implemented in a functional manner to avoid any unintended modification
+ */
+public class Channel {
+
+  private static final String ROOT_CHANNEL = "/root";
+  private static final String DELIMITER = "/";
+  // Predicate matching a String with the protocol regex of a channel
+  private static final Predicate<String> CHANNEL_VALIDATOR =
+      Pattern.compile("^/root(/[^/]+)*$").asPredicate();
+
+  /** Root channel, every client is subscribed to it. */
+  public static final Channel ROOT = new Channel();
+
+  public static Channel newChannel(String value) {
+    // Avoid creating duplicate instances
+    if (ROOT_CHANNEL.equals(value)) return ROOT;
+
+    // Remove the "/root/" part of the value as it is not relevant and does not need to be stored
+    String relevantPart = value.substring(ROOT_CHANNEL.length() + 1);
+    return new Channel(relevantPart.split(DELIMITER));
+  }
+
+  // ================ Object itself =================
+
+  // This list contains the different parts of the channel except the root as every channel has a
+  // root part
+  // So for the channel /root/lao_id/election_id, the elements will be [lao_id, election_id]
+  @NonNull private final List<String> elements;
+
+  private Channel(String... elements) {
+    this.elements = Arrays.asList(elements);
+
+    if (!CHANNEL_VALIDATOR.test(getAsString()))
+      throw new IllegalArgumentException(getAsString() + " is not a valid channel");
+  }
+
+  private Channel(Channel base, String subElement) {
+    List<String> newElems = new ArrayList<>(base.elements);
+    newElems.add(subElement);
+    this.elements = Collections.unmodifiableList(newElems);
+
+    if (!CHANNEL_VALIDATOR.test(getAsString()))
+      throw new IllegalArgumentException(getAsString() + " is not a valid channel");
+  }
+
+  /**
+   * Return a channel that is this channel with one more element underneath
+   *
+   * @param subElement element to append to this channel
+   * @return a new Channel object
+   */
+  public Channel sub(String subElement) {
+    return new Channel(this, subElement);
+  }
+
+  /** @return true if this channel is of the form '/root/lao_id' */
+  public boolean isLaoChannel() {
+    return elements.size() == 1;
+  }
+
+  /** @return true if this channel could be an election channel */
+  public boolean isElectionChannel() {
+    return elements.size() == 2;
```

Comment:
It could be a number of things though. Not sure the logic here is very robust.
I'd at least put a big warning sign 😅

Reply:
I did think at naming this `canBeElectionChannel` 
But this is the kind of artifact of the past that we must keep, at least for now.

This is how it was done before, and I don't see and robust way to do the check with the current codebase.

Reply:
I'd just document it as a javadoc warning or remark that this is brittle and needs improvement. Maybe deprecation warning?